bittide/src/Bittide/Node.hs

```diff
@@ -191,3 +210,112 @@ managementUnit (ManagementConfig scatterConfig gatherConfig peConfig) linkIn nod
   (vecToTuple -> (nmuM2S0, nmuM2S1), nodeM2Ss) = splitAtI rest
   (_, nmuM2Ss) = toSignals (processingElement peConfig) (pure $ JtagIn low low low, nmuS2Ms)
   nmuS2Ms = suS2M :> guS2M :> nmuS2M0 :> nmuS2M1 :> nodeS2Ms
+
+managementUnitC ::
+  forall dom nodeBusses.
+  ( HiddenClockResetEnable dom
+  , CLog 2 (nodeBusses + NmuInternalBusses) <= 30
+  ) =>
+  -- |
+  -- ( Configures all local parameters
+  -- , Incoming 'Bittide.Link'
+  -- , Incoming @Vector@ of master busses
+  -- )
+  ManagementConfig nodeBusses ->
+  Circuit
+    (CSignal dom (DataLink 64))
+    ( CSignal dom (DataLink 64)
+    , Vec nodeBusses (Wishbone dom 'Standard (NmuRemBusWidth nodeBusses) (Bytes 4))
+    )
+managementUnitC (ManagementConfig scatterConfig gatherConfig peConfig) = circuit $ \linkIn -> do
+  jtag <- idleSource -< ()
+  peWbs <- processingElement peConfig -< jtag
+  ([wbScatCal, wbScat, wbGathCal, wbGu], nmuWbs) <- splitAtC d4 -< peWbs
+  linkOut <- gatherUnitWbC gatherConfig -< (wbGu, wbGathCal)
+  scatterUnitWbC scatterConfig -< (linkIn, wbScat, wbScatCal)
+  idC -< (linkOut, nmuWbs)
+
+-- These functions should be added to `clash-protocols`, there is a PR for this:
+-- https://github.com/clash-lang/clash-protocols/pull/116
+-- And a bittide-hardware issue:
+-- https://github.com/bittide/bittide-hardware/issues/645
+-- Append two separate vectors of the same circuits into one vector of circuits
```

Comment:
The PR on `clash-protocols` is still in draft, should that be merged in before this PR or do we merge this first, given this PR is open?

Reply:
Let's not block this PR on that, there's a tracking issue and I _intend_ to get back to that in the near future :)